modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -1048,6 +1048,20 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
       }
     }
 
+    /**
+      * Should be used when calculating schema for a structure.
+      *
+      * See https://github.com/disneystreaming/smithy4s/issues/1296 for details.
+      */
+    def fields: List[Field] = fieldsInternal(hintsExtractor = hints)
+
+    /**
+      * Should be used only on the call site 
+      * of the trait application where there is no need to call `unfoldTrait` for every hint of the trait. 
+      */
+    def getFieldsPlain: List[Field] =
```

Comment:
**thought:** this could be a different type which would make a mistake impossible. Question is, whether it is worth the effort.